react/features/video-quality/subscriber.js

```diff
@@ -193,43 +193,114 @@ function _updateReceiverVideoConstraints({ getState }) {
     const maxFrameHeight = Math.min(maxReceiverVideoQuality, preferredVideoQuality);
     const { remoteScreenShares } = state['features/video-layout'];
     const { visibleRemoteParticipants } = state['features/filmstrip'];
+    const { flags: { sourceNameSignaling } } = state['features/base/config'];
+    const tracks = state['features/base/tracks'];
 
     const receiverConstraints = {
         constraints: {},
         defaultConstraints: { 'maxHeight': VIDEO_QUALITY_LEVELS.NONE },
         lastN,
-        onStageEndpoints: [],
-        selectedEndpoints: []
+        ...sourceNameSignaling ? { prioritizedSources: [] } : { onStageEndpoints: [] },
+        ...sourceNameSignaling ? { selectedSources: [] } : { selectedEndpoints: [] }
     };
 
-    // Tile view.
-    if (shouldDisplayTileView(state)) {
-        if (!visibleRemoteParticipants?.size) {
-            return;
+    if (sourceNameSignaling) {
+        const participantIdToTrackSourceName = tracks.reduce((acc, { jitsiTrack, participantId }) => {
```

Comment:
Will it be better if this state is moved into the Redux store? Perhaps we can have an object to store all remote video tracks and another for the audio tracks? The track state will then update on the `SET_REMOTE_PARTICIPANTS` event.